### PR TITLE
Make synctex an optional dependency

### PR DESCRIPTION
**Pull request by oz123**
This is still work in progress. The code still needs to be disabled
in the C code.
Comments are welcome, I am a total idiot when it comes to autotools :-P.

### CODE REVIEW

**Comment by corvus1** [comment]
The patch from here https://github.com/oz123/mate-de-gentoo/blob/master/app-text/atril/files/atril-1.23.2-Make-synctex-optional.patch seems to work fine.

**Comment by rbuj** [comment]
@oz123 could you update this pull request in order to review it? Rebase your local branch with remote master & apply new changes if required.

**Comment by oz123** [comment]
@rbuj Thanks for getting back to me on this. Sure, I can do this.

File: configure.ac [hunk 1]
```diff
@@ -361,15 +361,25 @@ AC_SUBST([GIRTYPELIBDIR])
...
+  if test "x$has_synctex" = "xno"; then
+          AC_MSG_ERROR("SyncTeX support is disabled since library version $SYNCTEX_REQUIRED or newer not found")
+  fi
+  AC_SUBST(SYNCTEX_LIBS)
+  AC_SUBST(SYNCTEX_CFLAGS)
 fi
-AC_SUBST(SYNCTEX_LIBS)
-AC_SUBST(SYNCTEX_CFLAGS)
+
+AM_CONDITIONAL([ENABLE_SYNCTEX], [test "$enable_synctex" = "yes"])
```
Comment (rbuj) [inline configure.ac:381]: The ENABLE_SYNCTEX conditional is defined but it's not used.
```diff
--- a/libdocument/Makefile.am
+++ b/libdocument/Makefile.am
@@ -102,11 +102,14 @@ libatrildocument_la_CPPFLAGS = \
 
 libatrildocument_la_CFLAGS = \
        $(LIBDOCUMENT_CFLAGS)                   \
-       $(SYNCTEX_CFLAGS)                       \
        $(WARN_CFLAGS)                          \
        $(DISABLE_DEPRECATED)                   \
        $(AM_CFLAGS)
 
+if ENABLE_SYNCTEX
+libatrildocument_la_CFLAGS += $(SYNCTEX_CFLAGS)
+endif
+
 libatrildocument_la_LDFLAGS = \
        -version-info $(EV_DOCUMENT_LT_VERSION_INFO)    \
        -no-undefined   \
@@ -116,9 +119,11 @@ libatrildocument_la_LDFLAGS = \
 libatrildocument_la_LIBADD = \
        $(LIBDOCUMENT_LIBS)     \
        $(GMODULE_LIBS)         \
-       $(SYNCTEX_LIBS)         \
        $(ZLIB_LIBS)
 
+if ENABLE_SYNCTEX
+libatrildocument_la_LIBADD += $(SYNCTEX_LIBS)
+endif
 
 BUILT_SOURCES =                        \
        ev-document-type-builtins.c     \
```
https://www.gnu.org/software/automake/manual/html_node/Usage-of-Conditionals.html

File: configure.ac [hunk 2]
```diff
@@ -361,15 +361,25 @@ AC_SUBST([GIRTYPELIBDIR])
 AM_CONDITIONAL([ENABLE_INTROSPECTION],[test "$enable_introspection" = "yes"])
 
 dnl ================== libsynctex ===========================================
+AC_ARG_ENABLE([synctex],
+  [AS_HELP_STRING([--disable-synctex], [Disable support for synctex])],
+  [],
+  [enable_synctex=yes])
```
Comment (rbuj) [inline configure.ac:367]: only two values are allowed: "yes", "no"
```diff
--- a/configure.ac
+++ b/configure.ac
@@ -363,9 +363,11 @@ AM_CONDITIONAL([ENABLE_INTROSPECTION],[test "$enable_introspection" = "yes"])
 dnl ================== libsynctex ===========================================
 AC_ARG_ENABLE([synctex],
   [AS_HELP_STRING([--disable-synctex], [Disable support for synctex])],
-  [],
-  [enable_synctex=yes])
-
+  [case "${enableval}" in
+    yes) debug=true ;;
+    no)  debug=false ;;
+    *) AC_MSG_ERROR([bad value ${enableval} for --disable-synctex or --enable-synctex]) ;;
+  esac],[enable_synctex=yes])
 
 if test "$enable_synctex" = "yes"; then
   AC_DEFINE([ENABLE_SYNCTEX],[1],[Define if synctex support is enabled])
```
Test before:
```
$ ./autogen.sh --prefix=/usr --enable-synctex=ASDF
<cut>
    Synctex enabled.....:    ASDF
<cut>
```
Test after:
```
$ ./autogen.sh --prefix=/usr --enable-synctex=ASDF
<cut>
configure: error: bad value ASDF for --disable-synctex or --enable-synctex
```

**Comment by rbuj** [comment]
Fix -Wunused-function warnings:
```
ev-document.c:390:1: warning: ‘_ev_document_support_synctex’ defined but not used [-Wunused-function]
  390 | _ev_document_support_synctex (EvDocument *document)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
--
ev-view.c:5682:1: warning: ‘highlight_forward_search_results’ defined but not used [-Wunused-function]
 5682 | highlight_forward_search_results (EvView *view, cairo_t *cr, int page)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Test
```
$ ./autogen.sh --prefix=/usr --disable-synctex && make &> make.log && grep -A 2 unused-function make.log
```

**Comment by oz123** [comment]
@rbuj I believe all synctex relevant `defined but not used` warnings are now fixed.

File: configure.ac (hunk 2 above)
Comment (oz123) [reply configure.ac]: I implemented your suggested patch in a slightly different way, but the meaning is the same.

File: configure.ac (hunk 1 above)
Comment (oz123) [reply configure.ac:384]: I added your patch here too.

**Comment by rbuj** [comment]
The build log has a new warning:
```
$ ./autogen.sh --prefix=/usr --disable-synctex --enable-compile-warnings=maximum && make &> make.log
$ grep -A 2 Wunused-function make.log 
ev-view.c:5631:1: warning: ‘draw_rubberband’ defined but not used [-Wunused-function]
 5631 | draw_rubberband (EvView             *view,
      | ^~~~~~~~~~~~~~~
```

**Comment by oz123** [comment]
@rbuj thanks for the quick response. I saw this warning, but I was not sure it's related to synctex. Hence, my comment above.

**Comment by rbuj** [comment]
@oz123 The `draw_rubberband` function is only used at libview/ev-view.c:5682
https://github.com/mate-desktop/atril/blob/37d80d4c862cbd03525fe4393f09b92d1dff8f56/libview/ev-view.c#L5682
then you can include the code below within the block `#ifdef ENABLE_SYNCTEX` 
https://github.com/mate-desktop/atril/blob/37d80d4c862cbd03525fe4393f09b92d1dff8f56/libview/ev-view.c#L5630-L5658
Test
```
$ git grep -n draw_rubberband
libview/ev-view.c:5631:draw_rubberband (EvView             *view,
libview/ev-view.c:5682:         draw_rubberband (view, cr, &view_rectangle, alpha);
```

**Comment by rbuj** [comment]
Do not remove find highlight:
```patch
diff --git a/libview/ev-view-private.h b/libview/ev-view-private.h
index 4f6179e..e951572 100644
--- a/libview/ev-view-private.h
+++ b/libview/ev-view-private.h
@@ -125,9 +125,7 @@ struct _EvView {
 	GList **find_pages;
 	gint find_result;
 	gboolean jump_to_find_result;
-#ifdef ENABLE_SYNCTEX
 	gboolean highlight_find_results;
-#endif
 	EvDocumentModel *model;
 	EvPixbufCache *pixbuf_cache;
 	gsize pixbuf_cache_size;
diff --git a/libview/ev-view.c b/libview/ev-view.c
index 7c9281c..d25c3cf 100644
--- a/libview/ev-view.c
+++ b/libview/ev-view.c
@@ -177,10 +177,10 @@ static void       ev_view_remove_all                         (EvView
 static AtkObject *ev_view_get_accessible                     (GtkWidget *widget);
 
 /*** Drawing ***/
-#if ENABLE_SYNCTEX
 static void       highlight_find_results                     (EvView             *view,
 							      cairo_t            *cr,
 							      int                 page);
+#if ENABLE_SYNCTEX
 static void       highlight_forward_search_results           (EvView             *view,
 							      cairo_t            *cr,
 							      int                 page);
@@ -4076,10 +4076,8 @@ ev_view_draw (GtkWidget      *widget,
 
 		if (page_ready && should_draw_caret_cursor (view, i))
 			draw_caret_cursor (view, cr);
-#ifdef ENABLE_SYNCTEX
 		if (page_ready && view->find_pages && view->highlight_find_results)
 			highlight_find_results (view, cr, i);
-#endif
 		if (page_ready && EV_IS_DOCUMENT_ANNOTATIONS (view->document))
 			show_annotation_windows (view, i);
 		if (page_ready && view->focused_element)
@@ -5658,7 +5656,6 @@ draw_rubberband (EvView             *view,
 }
 
 
-#ifdef ENABLE_SYNCTEX
 static void
 highlight_find_results (EvView *view, cairo_t *cr, int page)
 {
@@ -5683,6 +5680,7 @@ highlight_find_results (EvView *view, cairo_t *cr, int page)
         }
 }
 
+#ifdef ENABLE_SYNCTEX
 static void
 highlight_forward_search_results (EvView *view, cairo_t *cr, int page)
 {
@@ -7525,9 +7523,7 @@ ev_view_find_search_changed (EvView *view)
 void
 ev_view_find_set_highlight_search (EvView *view, gboolean value)
 {
-#ifdef ENABLE_SYNCTEX
 	view->highlight_find_results = value;
-#endif
 	gtk_widget_queue_draw (GTK_WIDGET (view));
 }
 
```

**Comment by oz123** [comment]
> Do not remove find highlight:

As requested, I re-enabled `highlight_find_results`.

**Pull request merged**